### ISSUE TITLE
Unable to teleport players during PlayerLoginEvent

### ISSUE THREAD

**Issue by buchwasa** [pmmp/PocketMine-MP#2933]
### Issue description
When using PM 4.0, trying to use syncMovement on login will crash with getId() on null.
- Expected result: Player to login successfully
- Actual result: Crashes right off the bat

### Steps to reproduce the issue
1. Use a plugin that does something like teleporting players on login
2. Try to join the server
3. Profit!

### OS and versions

* PocketMine-MP: 4.0.0dev
* PHP: 7.3
* Server OS: Windows
* Game version: PE/Win10

### Crashdump, backtrace or other files

```
2019-05-22 [21:19:20] [Server thread/CRITICAL]: Error: "Call to a member function getId() on null" (EXCEPTION) in "src/pocketmine/network/mcpe/NetworkSession" at line 593
2019-05-22 [21:19:20] [Server thread/DEBUG]: #0 src/pocketmine/Player(2670): pocketmine\network\mcpe\NetworkSession->syncMovement(pocketmine\Player object, double 0, double 0, integer 2)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #1 src/pocketmine/Player(2684): pocketmine\Player->sendPosition(pocketmine\Player object, double 0, double 0, integer 2)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #2 plugins/UHCGames/src/uhcgames/EventListener(53): pocketmine\Player->teleport(pocketmine\math\Vector3 object)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #3 src/pocketmine/plugin/RegisteredListener(89): uhcgames\EventListener->handleLogin(pocketmine\event\player\PlayerLoginEvent object)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #4 src/pocketmine/event/Event(67): pocketmine\plugin\RegisteredListener->callEvent(pocketmine\event\player\PlayerLoginEvent object)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #5 src/pocketmine/Player(341): pocketmine\event\Event->call()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #6 src/pocketmine/network/mcpe/NetworkSession(169): pocketmine\Player->__construct(pocketmine\Server object, pocketmine\network\mcpe\NetworkSession object, pocketmine\PlayerInfo object, boolean 1)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #7 src/pocketmine/network/mcpe/NetworkSession(565): pocketmine\network\mcpe\NetworkSession->createPlayer()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #8 src/pocketmine/network/mcpe/handler/ResourcePacksSessionHandler(111): pocketmine\network\mcpe\NetworkSession->onResourcePacksDone()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #9 src/pocketmine/network/mcpe/protocol/ResourcePackClientResponsePacket(62): pocketmine\network\mcpe\handler\ResourcePacksSessionHandler->handleResourcePackClientResponse(pocketmine\network\mcpe\protocol\ResourcePackClientResponsePacket object)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #10 src/pocketmine/network/mcpe/NetworkSession(325): pocketmine\network\mcpe\protocol\ResourcePackClientResponsePacket->handle(pocketmine\network\mcpe\handler\ResourcePacksSessionHandler object)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #11 src/pocketmine/network/mcpe/NetworkSession(295): pocketmine\network\mcpe\NetworkSession->handleDataPacket(pocketmine\network\mcpe\protocol\ResourcePackClientResponsePacket object)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #12 src/pocketmine/network/mcpe/RakLibInterface(158): pocketmine\network\mcpe\NetworkSession->handleEncoded(string x.c.`a`...E..)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #13 vendor/pocketmine/raklib/src/server/ServerHandler(102): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(integer 0, raklib\protocol\EncapsulatedPacket object, integer 0)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #14 src/pocketmine/network/mcpe/RakLibInterface(98): raklib\server\ServerHandler->handlePacket()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #15 vendor/pocketmine/snooze/src/SleeperHandler(123): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #16 vendor/pocketmine/snooze/src/SleeperHandler(85): pocketmine\snooze\SleeperHandler->processNotifications()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #17 src/pocketmine/Server(1813): pocketmine\snooze\SleeperHandler->sleepUntil(double 1558577960.8901)
2019-05-22 [21:19:20] [Server thread/DEBUG]: #18 src/pocketmine/Server(1339): pocketmine\Server->tickProcessor()
2019-05-22 [21:19:20] [Server thread/DEBUG]: #19 src/pocketmine/PocketMine(264): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string C:\Users\Drew\Documents\PM\, string C:\Users\Drew\Documents\PM\plugins\)
```

**Comment by dktapps** [comment]
the player has not yet been assigned to their own network session when this happens.

**Comment by dktapps** [comment]
The core problem here is that this packet is trying to be sent in the first place. Since the StartGamePacket hasn't been sent when this happens, it shouldn't be trying to send position in the first place.

**Comment by dktapps** [comment]
This was fixed by 893f7cb6ef89fa4350febee22aa949eb35e68101.

**Comment by Unickorn** [comment]
Not sure if this should go here, but since then teleporting players to a different world still causes crashes.

```
Error: Call to a member function getWorld() on null
File: pmsrc/src/network/mcpe/NetworkSession
Line: 945
Type: Error

Backtrace:
#0 pmsrc/src/player/Player(656): pocketmine\network\mcpe\NetworkSession->onEnterWorld()
#1 pmsrc/src/entity/Entity(1334): pocketmine\player\Player->setPosition(object pocketmine\world\Position)
#2 pmsrc/src/entity/Entity(1400): pocketmine\entity\Entity->setPositionAndRotation(object pocketmine\world\Position, double 43.360076904297, double 89.897216796875)
#3 pmsrc/src/player/Player(2224): pocketmine\entity\Entity->teleport(object pocketmine\world\Position, NULL , NULL )
#4 plugins/Furnace_597.phar/src/Furnace/listener/events/PlayerLogin(29): pocketmine\player\Player->teleport(object pocketmine\world\Position)```
